PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/springcloud/deplolyment/SpringCloudDeploymentConfiguration.java

```diff
@@ -53,26 +52,28 @@ public SpringCloudDeploymentConfiguration(@NotNull Project project, @NotNull Con
             .build();
     }
 
-    protected SpringCloudDeploymentConfiguration(@NotNull SpringCloudDeploymentConfiguration source) {
-        super(source.getProject(), source.getFactory(), source.getName());
-        this.appConfig = JsonUtils.deepCopyWithJson(source.appConfig);
-    }
-
     @Override
     public void readExternal(Element element) throws InvalidDataException {
         super.readExternal(element);
-        XmlSerializer.deserializeInto(this.appConfig, element);
+        final Element appConfigElement = element.getChild("SpringCloudAppConfig");
```

Comment:
extract the encode/decode logic code in readExternal and writeExternal to another class and write a unit test against it.